webapp/sources/ldap-inventory/inventory-provisioning-web/src/main/scala/com/normation/inventory/provisioning/endpoint/InventoryFileWatcher.scala

```diff
@@ -211,44 +214,106 @@ class InventoryFileWatcher(
 }
 
 
-class ProcessFile(
-    inventoryProcessor: InventoryProcessor
-  , received          : File
-  , failed            : File
-  , waitForSig        : FiniteDuration
-  , sigExtension      : String // signature file extension, most likely ".sign"
-)(implicit scheduler: Scheduler) {
-  import scala.concurrent.duration._
+sealed trait WatchEvent
+object WatchEvent {
+  // file is written and so can be removed from Map
+  final case class End(file: File) extends WatchEvent
+  // file is currently modified, avoid processing
+  final case class Mod(file: File) extends WatchEvent
+}
+
 
```

Comment:
Here we replace a variable map + monix interuptible task + do on finish with a queue which takes two even: one for "file modified", and one for "file need cleaning". We keep the same logic for map+interrutible task (fiber are always interreruptibles, and should be lighter than monix task) and we enqueue a "can be clean" even when the process actually starts.